.github/workflows/ci.yml

```diff
@@ -62,13 +62,11 @@ jobs:
       - run: cargo nextest run --locked --workspace --exclude integration-tests --cargo-profile dev-release --profile ci ${{ matrix.flags }}
         env:
           RUST_BACKTRACE: short
-      - run: cargo llvm-cov report --profile dev-release --codecov --output-path unittests.json
-      - uses: codecov/codecov-action@eaaf4bedf32dbdc6b720b63067d99c4d77d6047d
+      - run: cargo llvm-cov report --profile dev-release --codecov --output-path codecov.json
+      - uses: actions/upload-artifact@v3
         with:
-          token: ${{ secrets.CODECOV_TOKEN }}
-          files: unittests.json
-          fail_ci_if_error: true
-          flags: unittests,${{ matrix.id }}
+          name: cov-${{ matrix.id }}-unit
```

Comment:
FWIW as far as I know you can upload multiple files to the same artifact: https://github.com/actions/upload-artifact#uploading-to-the-same-artifact

Which then in the final step would allow you to basically just download all the files at once in a single invocation of `download-artifact`

Reply:
or, well, if nothing else, they end up as a single artifact in the UI and all the files end up in a single directory in the last upload job, which gives more opportunity for iteration through the files.

Reply:
Oh I can even upload multiple times to the same name across jobs, interesting, thank you! :)
I had ended up downloading literally all the artifacts in order to work around this before 😅 

This is now done :)